tests/test_linxura.py

```diff
@@ -0,0 +1,147 @@
+"""Tests for Linxura quirks."""
+
+import pytest
+from zigpy.zcl.clusters.security import IasZone
+
+from tests.common import ClusterListener
+import zhaquirks
+import zhaquirks.linxura
+
+zhaquirks.setup()
+
+
+@pytest.mark.parametrize("quirk", (zhaquirks.linxura.button.LinxuraButton,))
+async def test_Linxura_button(zigpy_device_from_quirk, quirk):
```

Comment:
Also, this test isn't testing the quirk. It just checks that the `update_attribute` call is passed and increases coverage by making sure all lines from the quirk are called, but we never verify the outputs..
We also can't and don't need to test ZHA here.

A proper test should verify that all cluster listeners with `ZHA_SEND_EVENT` are called as expected.
You can use something like this (and make use of `parametrize`):
```python
    listener = mock.MagicMock()
    cluster.add_listener(listener)

    assert cluster_listener.zha_send_event.call_count == 1
    assert cluster_listener.zha_send_event.call_args == mock.call(
        "button_1_press", {"button": "button_3", "press_type": "press", "command_id": 10}
    )
```

Reply:
Modified based on suggestion, however I'm a newbie in python. Didn't found a way to use parametrize properly to optimise and shorten the testing code. Please help me on this. Thanks a lot.

Reply:
All good! I've pushed a small change regarding this here: https://github.com/zigpy/zha-device-handlers/pull/3392/commits/b81c52d8418eab6310577c4ab5134c73dccad662